models/model-news.js

```diff
@@ -35,8 +36,24 @@ exports.selectArticles = (topic, sort_by = 'created_at', order_by = 'DESC') => {
 
 }
 
-exports.selectArticleById = (article_id) => {
-    return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id]).then((results) => {
+exports.selectArticleById = (article_id, comment_count) => {
+    let articleQuery = `SELECT articles.* `
+
+
+    if (comment_count !== undefined) {
+        articleQuery += ` 
+        , COUNT(*) AS comment_count 
+        FROM articles 
+        LEFT JOIN comments ON comments.article_id = articles.article_id
+        WHERE articles.article_id = $1 
+        GROUP BY articles.author, title, articles.article_id; `
+    } 
+    else {
+        articleQuery+= `FROM articles WHERE article_id = $1;`
+    }
```

Comment:
No if/else necessary, it should just always return a comment count now. This is essentially a refactor of an earlier endpoint!